lib/ansible/modules/packaging/os/yum.py

```diff
@@ -639,6 +639,42 @@ def local_envra(path):
                                header[rpm.RPMTAG_ARCH])
 
 
+def set_env_proxy(function):
+    def fetching(spec, module, conf_file, installroot):
```

Comment:
Out of curiosity, what's the motivation for making this a new (nested) function instead of adding the handling of the `yum.conf` proxy config items to `yum_base`?

Reply:
it does not matter, because in the methods of installation and update of the package, if the package name is specified as URI, then the external method - "fetch_url" is used, which works with the system settings of the proxy. And if system proxy is not set, or it's different from setted in yum.conf, than we need set back configuration to previous state. 

Because of that i wrote decorator with arguments, that temporary set or change system proxy environment, if in yum.conf set proxy variables, then package fetched with method  <code>def fetch_rpm_from_url(spec, module=None):</code>, and finaly we set previous state of system proxy environment

Reply:
Solid point about the specified URI, makes sense. Thanks. 👍

Reply:
This seems like it should be a context manager rather than a pseudo-decorator

Reply:
Note that it might also be easier to just include this logic with the existing fetch_rpm_from_url() function rather than using a context-manager.  I could see doing it either way.